.github/workflows/cmake.yml

```diff
@@ -50,41 +59,100 @@ jobs:
 
       - name: Build
         # Build your program with the given configuration
-        run: cmake --build ${{github.workspace}}/build --config ${{matrix.build}}
+        run: cmake --build ${{github.workspace}}/build --config ${{matrix.build}} -j
 
       - name: Test
+        shell: bash
         working-directory: ${{github.workspace}}/build
-        # Execute tests defined by the CMake configuration.
-        # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
-        # Use debug flag to show all exeucted tests
-        run: ctest --debug -C ${{matrix.build}}
+        env:
+          AZURE_BLOB_STORE_CONNECTION_STRING: "DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://127.0.0.1:10000/devstoreaccount1;"
+        run: |
+          # What we do here:
+          # - we create a test CZI file (using the CZIcmd executable)
+          # - we start Azurite (in the background)
+          # - we then upload the test CZI file to Azurite as a blob in the container "testcontainer" with name "testblob"
+          # - we then run the unit-tests - the environment variable AZURE_BLOB_STORE_CONNECTION_STRING is used to configure the Azure SDK based stream tests
+          # - finally, we kill the Azurite process
+          #
+          # find the CZIcmd executable (we just built it)
+          czicmd="$(find . \( -name CZIcmd -o -name CZIcmd.exe \) -print0 | xargs -0 realpath)"
+          mkdir -p azurite
+          cd azurite
+          # now use the CZIcmd executable to create a test CZI file
+          "$czicmd" --command CreateCZI --createbounds "C0:2T0:2" --generatorpixeltype Gray8 --compressionopts "zstd1:ExplicitLevel=2;PreProcess=HiLoByteUnpack" --createsubblocksize "1024x1024" -o test --bitmapgenerator default
+          # start Azurite in the background
+          azurite --inMemoryPersistence --silent &
+          # create a blob container "testcontainer"
+          az storage container create --name testcontainer  --connection-string "$AZURE_BLOB_STORE_CONNECTION_STRING"
+          # upload the test CZI file to the container
+          az storage blob upload --container-name testcontainer --file "./test.czi" --name testblob --connection-string "$AZURE_BLOB_STORE_CONNECTION_STRING"
+          cd ..
+          #"$czicmd" --command PrintInformation --source-stream-class azure_blob_inputstream --source 'account=libczirwtestdata;containername=testcontainer;blobname=testblob;connectionstring=DefaultEndpointsProtocol\=http\;AccountName\=devstoreaccount1\;AccountKey\=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw\=\=\;BlobEndpoint\=http://127.0.0.1:10000/devstoreaccount1\;'
```

Comment:
Can we reuse _AZURE_BLOB_STORE_CONNECTION_STRING_ here?

Reply:
We could... but we'd have to do the "libCZI-Uri-string-escapement" here (note that "=" and ";" need to be escaped)